Account for non-standard post type menu locations

eyesofjeremy opened the pull request:
Since a post type can be registered with a non-standard location, it seems reasonable to account for this by setting up a function to check for the post type's `show_in_menu` setting. If it's set to a specific location, the modified plugin will use that for the parent location.

At that point, the $_GET['post_type'] parameter may not be reflective of the correct post type, but the $_GET['page'] parameter is generated by the plugin, so we use that as a reference for generating the appropriate metabox.

This addresses issue #9. It has not been extensively tested yet.

eyesofjeremy committed:
Add link to non-standard post type edit location

- Use edit.php?post_type=$post_type as default, but allow two ways to modify this:
1) If the post type is registered with a string value for $post_type->show_in_menu, we should use that.
2) Add a filter hook ‘ptad_admin_parent’ to allow further customization.

eyesofjeremy committed:
Use custom callback for description page if post type's menu location is not the standard

- This is a first pass, and while it functions, it doesn’t seem like optimal code just yet. If the edit location is non-standard, then we can’t *necessarily* rely on the $post_type GET parameter to be correct. But the $page parameter is set within the plugin to include the post type in the parameter, so we are using that to set up the appropriate post type description settings page.

mrwweb commented:
Thanks for this, @eyesofjeremy. It's gonne be a week or two before I can get to this, but I am extremely appreciative and don't want to let this sit too long.

Would you be willing to share your post type registration code so I can test against your exact setup? I've never worked on a site quite like that, so I'd love to make sure I'm looking at the same thing you are.

eyesofjeremy commented:
Sure, here is a test case plugin that sets up the post types per my site's needs.

[gtt.1.8.ptad.zip](https://github.com/mrwweb/post-type-archive-descriptions/files/4488493/gtt.1.8.ptad.zip)

Let me know if you need anything else. Thanks!

eyesofjeremy committed:
Version bump for beta

eyesofjeremy committed:
Fix comment errors for ptad_admin_parent filter

eyesofjeremy committed:
Direct Admin Bar link to correct location

- If the post type has a customized show_in_menu location,
the admin bar link needs to reflect this.

eyesofjeremy committed:
Beta version bump

eyesofjeremy commented:
Hi there,

My client found an issue — the Admin Bar link showing up on the Archive page was not accounting for the changed address. I've added a commit which fixes this, plus fixing an incorrect comment on a new filter, and a couple of version bumps for tracking (using the beta version on the client's live site.)

Best,
Jeremy

mrwweb commented on post-type-archive-descriptions.php:
```diff
@@ -241,11 +271,29 @@ function ptad_editor_field( $args ) {
+ * Output settings pages for custom CPT setup
+ */
+function ptad_settings_custom() {
+	$page = $_GET['page'];
```
This method of determining the post type should work for both top-level and nested admin pages, so I'm tempted to remove the check and then just use the `$_GET[]` var. Any objections?

mrwweb commented on post-type-archive-descriptions.php:
```diff
@@ -108,13 +133,18 @@ function ptad_enable_pages() {
 
 		if( post_type_exists( $post_type ) ) {
 
+				// Check if post type has a particular parent menu location
```
extra indent on these four lines

mrwweb commented on post-type-archive-descriptions.php:
```diff
@@ -44,6 +44,31 @@ function ptad_get_post_types() {
+ * @param string $post_type name of post type for the page
+ * @return bool|string parent for settings page
+ */
+function ptad_settings_page_parent( $post_type, $show_in_menu = false ) {
```
`$show_in_menu` needs documentation. The name is a little odd in this context even though I realize it's coming straight from the WP naming conventions. It's tempting to give it a different name, but that might just confuse things further.

mrwweb commented on post-type-archive-descriptions.php:
```diff
@@ -322,10 +370,12 @@ function ptad_admin_bar_links( $admin_bar ) {
 			'id'    => 'wp-admin-bar-edit',
 			'title' => $link_text,
-			'href'  => admin_url( 'edit.php?post_type=' . $post_type . '&page=' . $post_type . '-description' )
+			'href'  => admin_url( $parent_page . '&page=' . $post_type . '-description' )
```
Just a note that the admin bar link (starting on Line 383) in the Admin (goes from description edit page to post type archive) is broken for nested post type menu items. Can be fixed later.

mrwweb reviewed (changes requested):
@eyesofjeremy Thank you so much again for submitting this, and sorry that it took me so long to get this. I appreciate your work to make this plugin more flexible and this will definitely get incorporated. I flagged a few issues I'd love to tweak. Do you want to make these changes or would you prefer I do it? Don't want to step on your toes, but also realize it took me forever to get to this. I'll accept the PR and tweak further myself if I don't hear from you in a week or so.

mrwweb commented:
@eyesofjeremy Thanks again for your contribution! v1.2.0 was just released including your fixes 😃

If you have a wordpress.org username, please let me know so I can add you as a contributor to the plugin!

eyesofjeremy commented:
Hi @mrwweb sorry I didn't get back to you earlier, looks like you had a couple more tweaks in there. Happy to be listed as a contributor :) username is jcdesign on WP.org.

mrwweb commented:
No problem! Just added you and it should show up shortly :)
